dbscan/structref.go

```diff
@@ -56,8 +72,8 @@ func getColumnToFieldIndexMap(structType reflect.Type) map[string][]int {
 			}
 
 			childType := field.Type
-			if field.Type.Kind() == reflect.Ptr {
-				childType = field.Type.Elem()
+			for childType.Kind() == reflect.Ptr {
+				childType = childType.Elem()
 			}
```

Comment:
Good catch. 
I found two more places there we should do the same thing:
https://github.com/georgysavva/scany/blob/2eb5f0fe37853b71416ef4b1892edce522176491/dbscan/structref.go#L98
https://github.com/georgysavva/scany/blob/d03fac1ef3c48f83eea39a48af58b17312a3a45a/dbscan/dbscan.go#L136-L142
I think we should move that change in a separate PR and cover it with tests properly